airbyte-integrations/connectors/source-trustpilot/source_trustpilot/source.py

```diff
@@ -0,0 +1,178 @@
+            self.start_datetime = MinMaxDatetime(self.start_datetime, options)
+
+        self._timezone = datetime.timezone.utc
+        self._parser = DatetimeParser()
+
+        if not self.start_datetime.datetime_format:
+            self.start_datetime.datetime_format = self.datetime_format
+
+        self._start_datetime_value = self.start_datetime.get_datetime(self.config)
+
+        self.cursor_field = InterpolatedString.create(self.cursor_field, options=options)
+
+        self._cursor_field_name = self.cursor_field.eval(self.config)
+        self._new_cursor_field_name = 'new_' + self._cursor_field_name
+        self._old_cursor_field_name = 'old_' + self._cursor_field_name
+
+    def _format_datetime(self, dt: datetime.datetime):
+        return self._parser.format(dt, self.datetime_format)
+
+    def parse_date(self, date: str) -> datetime.datetime:
+        return self._parser.parse(date, self.datetime_format, self._timezone)
+
+    def update_cursor(self, stream_slice: StreamSlice, last_record: Optional[Record] = None):
+        if last_record is not None:
+            if self._first_record_value is None:
+                tmp = last_record.get(self.cursor_field.eval(self.config))
+                self._first_record_value = self.parse_date(tmp)
+        elif stream_slice is not None:
+            tmp = stream_slice.get(self._new_cursor_field_name)
+            self._stream_slice_value = self.parse_date(tmp)
+
+    def get_stream_state(self) -> StreamState:
+        return {
+            self._old_cursor_field_name: self._format_datetime(
+                self._stream_slice_value if self._stream_slice_value is not None else self._start_datetime_value),
+
+            self._new_cursor_field_name: self._format_datetime(self._first_record_value if self._first_record_value is not None else (
+                self._stream_slice_value if self._stream_slice_value is not None else self._start_datetime_value
+            ))
+        }
+
+    def stream_slices(self, sync_mode: SyncMode, stream_state: Mapping[str, Any]) -> Iterable[StreamSlice]:
+        return [self.get_stream_state()]
+
+
+@dataclass
+class TrustpilotPaginator(DefaultPaginator):
+    """
+    Paginator for Trustpilot.
+    Copy of DefaultPaginator instead of request params logic:
+    1. Trustpilot API doesn't support `page_size` param, so it's removed from request params on the fly.
+    2. Trustpilot API `page` param should be omitted on first page and then should start first value 2,
+       so it's updated in request params on the fly.
+    """
+
+    def get_request_params(
+            self,
+            *,
+            stream_state: Optional[StreamState] = None,
+            stream_slice: Optional[StreamSlice] = None,
+            next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Mapping[str, Any]:
+        res = self._get_request_options(RequestOptionType.request_parameter)
+        del res['page_size']
```

Comment:
```suggestion
        del res['page_size']
```
The request breaks if you keep this param?

Reply:
It doesn't break request, but I wanted to remove it to have completely same requests to real website requests.
Need to discuss it a little bit more.